plugins/modules/storagebox_subaccount.py

```diff
@@ -0,0 +1,473 @@
+      readonly: true
+      comment: "Updated path and readonly mode"
+
+- name: Delete a subaccount
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      state: absent
+      username: "backup1"
+
+- name: Change password for a subaccount
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      state: present
+      username: "backup1"
+      password: "n3wSecur3Pass"
+"""
+
+RETURN = r"""
+created:
+  description: Whether a new subaccount was created.
+  type: bool
+  returned: always
+
+deleted:
+  description: Whether the subaccount was deleted.
+  type: bool
+  returned: always
+
+updated:
+  description: Whether the subaccount's configuration was updated (excluding password changes).
+  type: bool
+  returned: always
+
+password_updated:
+  description: Whether the subaccount's password was updated.
+  type: bool
+  returned: always
+
+created_subaccount:
+  description: The created subaccount object returned from the API.
+  type: raw
+  returned: when created is true
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+from ansible_collections.community.hrobot.plugins.module_utils.robot import (
+    BASE_URL,
+    ROBOT_DEFAULT_ARGUMENT_SPEC,
+    fetch_url_json,
+)
+
+
+def encode_data(data):
+    """Converts booleans to lowercase strings and filters out None values."""
+    return urlencode(
+        {
+            key: str(value).lower() if isinstance(value, bool) else value
+            for key, value in data.items()
+            if value is not None
+        }
+    )
+
+
+def create_subaccount(module, storagebox_id, subaccount):
+    url = "{0}/storagebox/{1}/subaccount".format(BASE_URL, storagebox_id)
+    res, error = fetch_url_json(
+        module,
+        url,
+        method="POST",
+        data=encode_data(subaccount),
+        headers={"Content-type": "application/x-www-form-urlencoded"},
+        accept_errors=[
+            "STORAGEBOX_NOT_FOUND",
+            "STORAGEBOX_SUBACCOUNT_LIMIT_EXCEEDED",
+            "STORAGEBOX_INVALID_PASSWORD",
+        ],
+        timeout=30000,  # this endpoint is stupidly slow
+    )
+    if not error:
+        return res
+
+    if error == "STORAGEBOX_INVALID_PASSWORD":
+        module.fail_json(
+            msg="Creation of user with homedirectory '{0}' has an invalid password (says Hetzner)".format(
+                subaccount.get("homedirectory", "<missing>"),
+            )
+        )
+    if error == "STORAGEBOX_SUBACCOUNT_LIMIT_EXCEEDED":
+        module.fail_json(msg="Subaccount limit exceeded")
+
+    module.fail_json(
+        msg="Unknown error: {0} for username '{1}' and homedirectory '{2}'".format(
+            error,
+            subaccount.get("username", "<missing>"),
+            subaccount.get("homedirectory", "<missing>"),
+        )
+    )  # pragma: no cover
+
+
+class Subaccount:
```

Comment:
The collection still supports Python 2.7, so it should say:
```suggestion
class Subaccount(object):
```